Source/VMware.vSphereDSC/DSCResources/VDSwitchVMHost.ps1

```diff
@@ -0,0 +1,251 @@
+
+            $distributedSwitch = $this.GetDistributedSwitch()
+            $vmHosts = $this.GetVMHosts()
+
+            if ($null -eq $distributedSwitch) {
+                # If the Distributed Switch is $null, it means that Ensure is 'Absent' and the Distributed Switch does not exist.
+                return $true
+            }
+
+            return !$this.ShouldUpdateVMHostsInDistributedSwitch($vmHosts, $distributedSwitch)
+        }
+        finally {
+            $this.DisconnectVIServer()
+        }
+    }
+
+    [VDSwitchVMHost] Get() {
+        try {
+            $result = [VDSwitchVMHost]::new()
+
+            $this.ConnectVIServer()
+            $this.EnsureConnectionIsvCenter()
+
+            $distributedSwitch = $this.GetDistributedSwitch()
+            $vmHosts = $this.GetVMHosts()
+
+            $result.Server = $this.Connection.Name
+            $result.VMHostNames = $vmHosts | Select-Object -ExpandProperty Name
+            $result.Ensure = $this.Ensure
+
+            if ($null -eq $distributedSwitch) {
+                # If the Distributed Switch is $null, it means that Ensure is 'Absent' and the Distributed Switch does not exist.
+                $result.VdsName = $this.Name
+            }
+            else {
+                $result.VdsName = $distributedSwitch.Name
+            }
+
+            return $result
+        }
+        finally {
+            $this.DisconnectVIServer()
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the Distributed Switch with the specified name from the server if it exists.
+    If the Distributed Switch does not exist and Ensure is set to 'Absent', $null is returned.
+    Otherwise it throws an exception.
+    #>
+    [PSObject] GetDistributedSwitch() {
+        if ($this.Ensure -eq [Ensure]::Absent) {
+            $distributedSwitch = Get-VDSwitch -Server $this.Connection -Name $this.VdsName -ErrorAction SilentlyContinue
+            return $distributedSwitch
+        }
+        else {
+            try {
+                $distributedSwitch = Get-VDSwitch -Server $this.Connection -Name $this.VdsName -ErrorAction Stop
+                return $distributedSwitch
+            }
+            catch {
+                throw "Could not retrieve Distributed Switch $($this.VdsName). For more information: $($_.Exception.Message)"
+            }
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the VMHosts with the specified names from the server if they exist.
+    For every VMHost that does not exist, a warning message is shown to the user without throwing an exception.
+    #>
+    [array] GetVMHosts() {
+        $vmHosts = @()
+
+        foreach ($vmHostName in $this.VMHostNames) {
+            $vmHost = Get-VMHost -Server $this.Connection -Name $vmHostName -ErrorAction SilentlyContinue
+            if ($null -eq $vmHost) {
+                Write-WarningLog -Message "The passed VMHost {0} was not found and it will be ignored." -Arguments @($vmHostName)
```

Comment:
Essentially this means the described state is incomplete for the system. It's interesting what the "correct" behavior in this case.

Warning and continue seems ok in this case
Validate first and do not apply  is another option.

I'm wondering is there a DSC mechanism that enables user to choose what to happens if configuration is incomplete, or it is up to the resource to manage the behvior with parameters?

Reply:
Well I have not found a common mechanism that handles this kind of behaviour so at the moment I think each DSC Resource should decide what the behaviour should be based on the use case.